mithril-test-lab/mithril-devnet/mkfiles/mkfiles-init.sh

```diff
@@ -65,20 +65,21 @@ if ! mkdir -p "${ARTIFACTS_DIR}"; then
   exit
 fi
 
+echo ">> Creating artifacts directory..."
 # Download cardano-cli & cardano-node if enabled (default: yes)
 if [[ "$SKIP_CARDANO_BIN_DOWNLOAD" != "true" ]]; then
   echo ">> Downloading cardano-cli & cardano-node..."
-  curl -sL ${CARDANO_BINARY_URL} --output cardano-bin.tar.gz
-  tar xzf cardano-bin.tar.gz ./bin/cardano-cli ./bin/cardano-node && mv ./bin/cardano-{cli,node} . && rm -rf ./bin  || tar xzf cardano-bin.tar.gz ./cardano-cli ./cardano-node
+  curl -sL "${CARDANO_BINARY_URL}" --output cardano-bin.tar.gz
+  echo ">> Extracting cardano-cli & cardano-node..."
+  tar xzf cardano-bin.tar.gz ./bin -C "${ARTIFACTS_DIR}/" || (mkdir -p "${ARTIFACTS_DIR}/bin" && tar --strip-components=1 -C "${ARTIFACTS_DIR}/bin" -xzf cardano-bin.tar.gz)
   rm -f cardano-bin.tar.gz
 fi
 
-# And copy cardano-cli & cardano-node
-cp cardano-cli ${ARTIFACTS_DIR}/cardano-cli
-cp cardano-node ${ARTIFACTS_DIR}/cardano-node
+cp -r ./bin "${ARTIFACTS_DIR}/" 2>/dev/null || true
+rm -rf ./bin
```

Comment:
hey @Alenar 

actually this specific suggestion is crashing the tests now by using

```bash
tar xzf cardano-bin.tar.gz ./bin -C "${ARTIFACTS_DIR}/" || (mkdir -p "${ARTIFACTS_DIR}/bin" && tar --strip-components=1 -C "${ARTIFACTS_DIR}/bin" -xzf cardano-bin.tar.gz)
```

and removing the decompression part:

```bash
cp -r ./bin "${ARTIFACTS_DIR}/" 2>/dev/null || true
rm -rf ./bin
```

here you have some logs:

```bash
cargo build --release -p mithril-aggregator -p mithril-signer -p mithril-client-cli -p mithril-relay && cargo run -p mithril-end-to-end -- -vvvvvvvvvv --bin-directory target/release/ --work-directory=$(PWD)/artifacts --devnet-scripts-directory=./mithril-test-lab/mithril-devnet --cardano-node-version 9.1.0
    Finished `release` profile [optimized] target(s) in 0.20s
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.16s
     Running `target/debug/mithril-end-to-end -vvvvvvvvvv --bin-directory target/release/ --work-directory=/Users/falcucci/txpipe/mithril/artifacts --devnet-scripts-directory=./mithril-test-lab/mithril-devnet --cardano-node-version 9.1.0`
Aug 20 17:20:56.065 INFO Bootstrapping the Devnet, script: /Users/falcucci/txpipe/mithril/mithril-test-lab/mithril-devnet/devnet-mkfiles.sh
tar: -C: Not found in archive
tar: /Users/falcucci/txpipe/mithril/artifacts/devnet: Not found in archive
tar: Error exit delayed from previous errors.
/Users/falcucci/txpipe/mithril/mithril-test-lab/mithril-devnet/mkfiles/mkfiles-cardano.sh: line 72: ./bin/cardano-cli: No such file or directory
Aug 20 17:20:59.689 INFO Starting the Devnet, script: /Users/falcucci/txpipe/mithril/artifacts/devnet/start-cardano.sh
Error: Failed to start the devnet

Caused by:
    No such file or directory (os error 2)
```

Reply:
the previous approach was working just fine,

actually I would say it's much clearer in terms of readability the previous way isolating those steps by:

```bash
tar xzf cardano-bin.tar.gz ./bin || (mkdir -p ./bin && tar --strip-components=1 -C ./bin -xzf cardano-bin.tar.gz)
```

and then using 

```bash
cp -r ./bin "${ARTIFACTS_DIR}/" 2>/dev/null || true
rm -rf ./bin
```

Reply:
but this is a take it or leave it, I can also apply your suggestion once it stops crashing and has no-warnings.

Reply:
I checked it on my side and the error is due to argument order on the first tar command, the `-C  "${ARTIFACTS_DIR}"` must come before the `./bin` since it only apply to the arguments after it (my bad :bow: ).

So the tar line should be:
```bash
tar xzf cardano-bin.tar.gz -C "${ARTIFACTS_DIR}/" ./bin || (mkdir -p "${ARTIFACTS_DIR}/bin" && tar --strip-components=1 -C "${ARTIFACTS_DIR}/bin" -xzf cardano-bin.tar.gz)
```

This should work like that, if there's still errors even with this fix you can revert to your previous proposition so we can merge.

Reply:
thank you @Alenar 

done.